pkg/arvo/app/chat-pull-hook.hoon

```diff
@@ -0,0 +1,260 @@
+/-  *invite-store, hook=chat-pull-hook
+/+  default-agent, verb, dbug, store=chat-store, lib=chat-hooks
+|%
++$  card  card:agent:gall
++$  versioned-state
+  $%  [%0 state-0]
+  ==
++$  state-0
+  $:  tracking=(map path ship)
+  ==
+--
+::
+=|  [%0 state-0]
+=*  state  -
+::
+%-  agent:dbug
+%+  verb  |
+::
+^-  agent:gall
+|_  bol=bowl:gall
++*  this  .
+    def  ~(. (default-agent this %|) bol)
+::
+++  on-init
+  ^-  (quip card _this)
+  :_  this
+  =/  inv  [our.bol %invite-store]
+  :~  [%pass / %agent inv %poke %invite-action !>([%create /chat])]
+      [%pass /invites %agent inv %watch /invitatory/chat]
+  ==
+::
+++  on-save  !>(state)
+++  on-load
+  |=  vax=vase
+  =/  old  !<(versioned-state vax)
+  ?-  -.old
+    %0  [~ this(state old)]
+  ==
+::
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  ?.  ?=(%chat-pull-hook-action mark)
+    (on-poke:def mark vase)
+  =/  act  !<(action:hook vase)
+  ?-  -.act
+      %add
+    ?>  (team:title our.bol src.bol)
+    ?:  (~(has by tracking) path.act)  [~ this]
+    =.  tracking  (~(put by tracking) path.act ship.act)
+    ?.  ask-history.act
+      =/  chat-path  [%mailbox path.act]
+      :_  this
+      [%pass chat-path %agent [ship.act %chat-push-hook] %watch chat-path]~
+    =/  mailbox=(unit mailbox:store)  (chat-scry:lib bol path.act)
+    =/  backlog=path
+      :-  %backlog
+      %+  weld  path.act
+      ?~(mailbox /0 /(scot %ud (lent envelopes.u.mailbox)))
+    :_  this
+    :~  [%pass backlog %agent [ship.act %chat-push-hook] %watch backlog]
+        :*  %give
+            %fact
+            ~[/tracking]
+            %chat-pull-hook-update
+            !>([%tracking tracking])
+    ==  ==
+  ::
+      %remove
+    ::  XX chat-hook had a path for reading the ship out of the path if not
+    ::  present in the tracking map. Why is this helpful or necessary?
+    ::  If it's not in the map, what's the effect of removing it?
+    ::  Canceling subscriptions that shouldn't exist?
```

Comment:
Yes, the idea was to allow cancellation of subscriptions that stuck around when they shouldn’t have. Feel free to ignore that code path and not write it.